python/ray/serve/autoscaling_metrics.py

```diff
@@ -102,19 +139,30 @@ def window_average(
             The average of all the datapoints for the key on and after time
             window_start_timestamp_s, or None if there are no such points.
         """
-        datapoints = self.data[key]
-
-        idx = bisect.bisect(
-            a=datapoints,
-            x=TimeStampedValue(
-                timestamp=window_start_timestamp_s, value=0  # dummy value
-            ),
-        )
-        points_after_idx = datapoints[idx:]
+        points_after_idx = self._get_datapoints(key, window_start_timestamp_s)
 
         if do_compact:
             self.data[key] = points_after_idx
 
         if len(points_after_idx) == 0:
             return
         return sum(point.value for point in points_after_idx) / len(points_after_idx)
+
+    def max(self, key: str, window_start_timestamp_s: float):
+        """Perform a max operation for metric `key`
+
+        Args:
+            key(str): the metric name.
```

Comment:
```suggestion
            key (str): the metric name.
```

Reply:
I think we don't add space in most of docstrings in the code. :) Please let me know if we are under any style change.

Reply:
Hmm, yeah we should make a more consistent style guide for this. I suggested it because in some places in this change, [there's a space](https://github.com/ray-project/ray/pull/24892/files#diff-d59bb9bd79ea8faac34971edc10b2ceab555f660c0c7a663d5ec9d86fa416886R83) before the type and in others (like here) there's not.

If there's a lot of places to edit for this, we don't need to spend too much time on this. Feel free to resolve this conversation.

Reply:
I will make further changes until we have consistent style, i think code base is not clean for this consistency.